### ISSUE TITLE
VIM gets stuck in mouse pressed state if popup is visible

### ISSUE THREAD

**Issue by noscript** [vim/vim#8318]
**Describe the bug**
Sometimes after a mouse click VIM goes into `VISUAL` mode that cannot be canceled anyhow unless clicked again. It takes quite a few clicks to reproduce.

This only happens if popup is visible and while VIM is blocked by external process.

**To Reproduce**
1. Run VIM:
```
vim --clean -g \
  +'call timer_start(1000, {->system("sleep 1")}, #{repeat: -1})' \
  +'call popup_create("hello", {})' \
  README.txt
```
2.  Start clicking at different places with at least a second between clicks.
3. VIM goes to SELECT mode. Pressing `Esc` does nothing. Moving mouse changes the selected area.
4. Click again to go unstuck the mouse.

**Expected behavior**
VIM does not go into eternal mouse pressed state.

**Screenshots**

https://user-images.githubusercontent.com/717109/120637982-3b016980-c478-11eb-8d2c-ce6959e9d5aa.mp4

**Environment**
 - Vim 8.2.2914
 - OS: Ubuntu 20.04
 - Terminal: GUI

**Comment by brammool** [comment]
I can reproduce it.  Looks like the sleep causes some mouse events to go missing and make it look like a double click.  That is hard to reproduce and I'm not sure how to fix it.
What I can fix is that when pressing Esc the Visual mode doesn't come back.

**Comment by noscript** [comment]
That's very unfortunate there is no proper fix. Even though it doesn't reproduce on every click I get struck by this all the time.

**Comment by brammool** [comment]
> That's very unfortunate there is no proper fix. Even though it doesn't
> reproduce on every click I get struck by this all the time.

You don't actually have a system('sleep 1') call in your code, right?
If you call a shell command Vim will be blocked and unresponsive, that
is the cause of the problem.  You can try to find another solution for
that.

-- 
Dogs must have a permit signed by the mayor in order to congregate in groups
of three or more on private property.
		[real standing law in Oklahoma, United States of America]

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by noscript** [comment]
>You don't actually have a system('sleep 1') call in your code, right?

No I don't have. It is there only to reproduce the bug.

I have a vim script function that apparently runs for long enough to trigger the problem. I will try to split the execution and use timers. Hopefully it helps.

**Comment by noscript** [comment]
I tried to split and distribute the execution as much as possible but I still struggle with the issue. I don't know what else to try. Splitting it even further will turn my code into an unmaintainable spaghetti mess.

I lowered the sleep time and timer frequency to make it is much easier to reproduce, basically this way it happens almost on every click:
```
vim --clean -g \
  +'call timer_start(100, {->system("sleep 0.1")}, #{repeat: -1})' \
  +'call popup_create("hello", {})' \
  README.txt
```

The problem doesn't happen if there are no popups visible.
